### ISSUE TITLE
[Bug] make sync fails locally 

### ISSUE THREAD

**Issue by shubhscoder** [ray-project/kuberay#1368]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/kuberay/issues) and found no similar issues.

### KubeRay Component

ray-operator

### What happened + What you expected to happen

**The bug:** When I ran make sync locally I got an error

```
test -s /vagrant/kuberay/ray-operator/bin/kustomize ||  GOBIN=/vagrant/kuberay/ray-operator/bin/kustomize/.. go install sigs.k8s.io/kustomize/kustomize/v3@v3.10.0
go: sigs.k8s.io/kustomize/kustomize/v3@v3.10.0 (in sigs.k8s.io/kustomize/kustomize/v3@v3.10.0):
        The go.mod file for the module providing named packages contains one or
        more exclude directives. It must not contain directives that would cause
        it to be interpreted differently than if it were the main module.
Makefile:117: recipe for target 'kustomize' failed
make: *** [kustomize] Error 1

```
**Expected beahvior**
make sync should work fine and synchronize the required yamls.

**Useful Information**

Go version : go version go1.19.12 linux/amd64
Kustomize install version : v3@v3.10.0

My guess is that it is related to this issue: https://github.com/kubernetes-sigs/kustomize/issues/3618#issuecomment-781521569

I could get around this issue by simply updating the kustomize version to 4.5.2 locally in the Makefile. Supporting comment for this: https://github.com/kubernetes-sigs/kustomize/issues/3618#issuecomment-1054010950

I think the fix is just updating the version of Kustomize. However, I am not confident of what other things are dependent on this (Github CI/CD, other builds, etc). So, opening this issue for investigation. 

### Reproduction script

Navigate to `kuberay/ray-operator`

Run command : `make sync`

or 

`make helm`

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

**Comment by kevin85421** [comment]
I took a look at this issue:

1. `kustomize` is not required for `helm`, so you can remove it. Then, you can run `make sync` successfully.
2. Although we can execute `make sync` successfully after (1), the command `make kustomize` will still fail with the same error message that you reported. You can upgrade the version of Kustomize. For example, the latest stable release is v5.1.1 ([link](https://github.com/kubernetes-sigs/kustomize)).

```makefile
KUSTOMIZE = $(shell pwd)/bin/kustomize
kustomize: ## Download kustomize locally if necessary.
	test -s $(KUSTOMIZE) ||  GOBIN=$(KUSTOMIZE)/.. go install sigs.k8s.io/kustomize/kustomize/v5@v5.1.1
```

**Comment by kevin85421** [comment]
@shubhscoder would you mind opening a PR based on https://github.com/ray-project/kuberay/issues/1368#issuecomment-1696777888? Thanks!

Btw, I am interested in why does Kustomize is still under maintenance after it was integrated into `kubectl`. Maybe only some parts of the functionalities are integrated?

**Comment by kevin85421** [comment]
This is solved by #1371.
